src/skin/legacyskinparser.cpp

```diff
@@ -304,19 +306,34 @@ QWidget* LegacySkinParser::parseSkin(QString skinPath, QWidget* pParent) {
 
     SkinManifest manifest = getSkinManifest(skinDocument);
 
-    // Apply SkinManifest attributes.
+    // Apply SkinManifest attributes by looping through the proto.
     for (int i = 0; i < manifest.attribute_size(); ++i) {
         const SkinManifest::Attribute& attribute = manifest.attribute(i);
         if (!attribute.has_config_key()) {
             continue;
         }
-        ConfigKey configKey = ConfigKey::parseCommaSeparated(
-            QString::fromStdString(attribute.config_key()));
 
         bool ok = false;
         double value = QString::fromStdString(attribute.value()).toDouble(&ok);
         if (ok) {
-            ControlObject::set(configKey, value);
+            ConfigKey configKey = ConfigKey::parseCommaSeparated(
+                    QString::fromStdString(attribute.config_key()));
+            // Set the specified attribute, possibly creating the control
+            // object in the process.
+            bool created = false;
+            // Update the value specified in the skin with whatever is in the
+            // saved config, if any.
+            value = m_pConfig->getValueString(configKey,
+                                              QString::number(value)).toDouble();
+            ControlObject* pControl = controlFromConfigKey(configKey,
+                                                           attribute.persist(),
+                                                           &created);
+            pControl->set(value);
```

Comment:
This circumstance is a little different -- the value should be set if the control was created, not just if the value is persisted.  This allows everything to work on first run, when the config doesn't have a value to load.  I also made sure to initialize the config value if it doesn't exist already.